pkg/cloud/services/wait/wait_test.go

```diff
@@ -55,30 +61,59 @@ func TestWaitForWithRetryable(t *testing.T) {
 			condition: func() (done bool, err error) {
 				return false, nil
 			},
-			retryableErrors: nil,
+			retryableErrors: retryableErrorCodes,
 			expectedError:   wait.ErrWaitTimeout,
 		},
 		{
 			name:    "error occurred in conditionFunc, returns actual error",
 			backoff: backoff,
 			condition: func() (done bool, err error) {
-				return false, errors.New("new error")
+				return false, nonRetryableError
+			},
+			retryableErrors: retryableErrorCodes,
+			expectedError:   nonRetryableError,
+		},
+		{
+			name:    "timed out in retryable error, returns the retryable error",
+			backoff: backoff,
+			condition: func() (done bool, err error) {
+				return false, retryableError
+			},
+			retryableErrors: retryableErrorCodes,
+			expectedError:   retryableError,
+		},
+		{
+			name:    "retryable err at first, non-retryable err after that, returns latest error",
+			backoff: backoff,
+			condition: func() (done bool, err error) {
+				if firstIteration {
+					firstIteration = false
+					return false, retryableError
+				}
+				firstIteration = false
+				return false, nonRetryableError
 			},
-			retryableErrors: nil,
-			expectedError:   errors.New("new error"),
+			retryableErrors: retryableErrorCodes,
+			expectedError:   nonRetryableError,
 		},
 		{
-			name:    "timed out in retryable error, returns actual error",
+			name:    "retryable err at first, no error after that, returns timeout error",
 			backoff: backoff,
 			condition: func() (done bool, err error) {
-				return false, errors.New("retryable error")
+				if firstIteration {
+					firstIteration = false
+					return false, retryableError
+				}
+				return false, nil
 			},
-			retryableErrors: []string{"retryable error"},
-			expectedError:   errors.New("retryable error"),
+			retryableErrors: retryableErrorCodes,
+			expectedError:   wait.ErrWaitTimeout,
 		},
```

Comment:
In my changes, I had added a test case for success after returning a retryable error as well, might be good to add that.